modules/tncIdSystem.js

```diff
@@ -0,0 +1,85 @@
+import { submodule } from '../src/hook.js';
+import { logInfo } from '../src/utils.js';
+
+const MODULE_NAME = 'tncId';
+const FALLBACK_TNC_PROVIDERID = 'c8549079-f149-4529-a34b-3fa91ef257d1';
+const FALLBACK_TNC_INSTANCE = '__tncPbjs';
+
+const loadRemoteTNCScript = (providerId) => {
+  try {
+    let tncOptions = {
+      autostart: true,
+      tcf: true,
+      getFiles: {
+        cacheid: true, // true
+        options: false, // true
+        callback: false, // true
+        prepare: false, // false
+        scripts: false // true
+      }
+    };
+    let t = window;
+    let n = t.document;
+    let c = FALLBACK_TNC_INSTANCE;
+    let i = providerId;
+    let d = tncOptions;
+
+    t[c] = t[c] || {}; t[c].providerId = i; t[c].options = t[c].options || d;
+    var f = 'ready'; t[c][f] = t[c][f] || function (z) {
+      t[c][f].q = t[c][f].q || []; t[c][f].q.push(z);
+      return t[c];
+    }; var s = n.createElement('script'); s.setAttribute('global', c); s.async = !0;
+    s.defer = !0; s.id = 'tnc_route'; s.src = 'https://js.tncid.app/route.js';
+    n.querySelector('head,body').appendChild(s);
```

Comment:
I think you should be using the built in `loadExternalScript()` method to load your script... @patmmccann or @dgirardi do you have thoughts on this?

Reply:
Thank you so much for your feedback.

We do not use loadExternalScript function or endpoint because of the particularity of our UserID script. We do not save UserID in cookies or localstorage but it is stored in its own namespace on a global window object.

As we would like to use a separate namespace for prebit script and we need to let our external script know where to look.

In modules/tncIdSystem.js:

    t[c] = t[c] || {}; t[c].providerId = i; t[c].options = t[c].options || d;
    var f = 'ready'; t[c][f] = t[c][f] || function (z) {
      t[c][f].q = t[c][f].q || []; t[c][f].q.push(z);
      return t[c];
    }; var s = n.createElement('script'); s.setAttribute('global', c); s.async = !0;
    s.defer = !0; s.id = 'tnc_route'; s.src = 'https://js.tncid.app/route.js';
    n.querySelector('head,body').appendChild(s);

Current part of the code creates a script tag with an attribute named "global" where the name of the namespace is stored for the use of an external loaded script.

Please let us know if the current approach may violate prebit best practices or if you have some better solution to propose.

Reply:
This isn't acceptable, perhaps you should just give publishers the location of your script and assume it's on page in this adapter?

Reply:
Hi @patmmccann, thank you for your feedback. 

We made changes to our TNCID generating script and we now use loadExternalScript util function for loading our external script.  

Hope it now satisfies all Prebid coding rules.

Reply:
Hi @smenzer ,

Thank you for your feedback. There are few examples of scripts with similar situations. Your precise indications are really useful.

We completely **removed** this part of code that creates script tag:

> t[c] = t[c] || {}; t[c].providerId = i; t[c].options = t[c].options || d;
> var f = 'ready'; t[c][f] = t[c][f] || function (z) {
>   t[c][f].q = t[c][f].q || []; t[c][f].q.push(z);
>   return t[c];
> }; var s = n.createElement('script'); s.setAttribute('global', c); s.async = !0;
> s.defer = !0; s.id = 'tnc_route'; s.src = 'https://js.tncid.app/route.js';
> n.querySelector('head,body').appendChild(s);

Now we use loadExternalScript function but only if client doesn't have our script already on page or if prebid script was executed first:

> const loadRemoteScript = (providerId) => {
>   return new Promise((resolve) => {
>     loadExternalScript('https://js.tncid.app/remote.js?ns=' + FALLBACK_TNC_INSTANCE + '&providerId=' + providerId, MODULE_NAME, resolve);
>   })
> }
> 
> const tncCallback = function (providerId, cb) {
>   let tncNS = '__tnc';
>   let promiseArray = [];
>   if (!window[tncNS] && !disableFallbackScript) {
>     tncNS = FALLBACK_TNC_INSTANCE;
>     promiseArray.push(loadRemoteScript(providerId));
>   }
> 
>   return Promise.all(promiseArray).then(() => waitTNCScript(tncNS)).then(cb).catch(() => cb(null));
> }

Moreover we added input param to disable load of fallback script for more control by publisher
> disableFallbackScript = (config && config.params && config.params.disableFallbackScript);

Reply:
@patmmccann are you ok with this approach?

Reply:
If they want to open source and lock that script similar, sure, otherwise we risk publishers interpreting it as vetted and endorsed by the project when in fact we haven't seen it and cannot warranty its contents

Reply:
ok that's what i thought, thanks @patmmccann